MantidPlot/src/FloatingWindow.cpp

```diff
@@ -39,6 +39,14 @@ FloatingWindow::FloatingWindow(ApplicationWindow *appWindow, Qt::WindowFlags f)
   // Instead, the ApplicationWindow->removeFloatingWindow() call takes care of
   // calling deleteLater().
   setAttribute(Qt::WA_DeleteOnClose, false);
+
+#ifdef Q_OS_MAC
+  // Work around to ensure that floating windows remain on top of the main
+  // application window, but below other applications on Mac
+  Qt::WindowFlags flags = windowFlags();
+  Qt::WindowFlags new_flags = flags | Qt::Tool;
```

Comment:
A few lines up (line 35) there is similar code for windows. They save the flags in `m_flags` which is then used in `FloatingWindow::event()` when there is a `WindowStateChange`. Would that be needed on osx as well?